### PR TITLE
Changes to support design time code generation in Mvc

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Related pr : https://github.com/aspnet/Mvc/pull/3190

File: src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -65,7 +65,8 @@ public override CodeGeneratorResult Generate()
                     var csharpCodeVisitor = CreateCSharpCodeVisitor(writer, Context);
 
                     new CSharpTypeMemberVisitor(csharpCodeVisitor, writer, Context).Accept(Tree.Chunks);
-                    new CSharpDesignTimeHelpersVisitor(csharpCodeVisitor, writer, Context).AcceptTree(Tree);
+                    CreateCSharpDesignTimeCodeVisitor(csharpCodeVisitor, writer, Context)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeGenerator.cs]: nit: can all fit on one line

File: src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpDesignTimeCodeVisitor.cs [hunk 2]
```diff
@@ -31,19 +28,11 @@ public class CSharpDesignTimeHelpersVisitor : CodeVisitor<CSharpCodeWriter>
...
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
-            _csharpCodeVisitor = csharpCodeVisitor;
+            CSharpCodeVisitor = csharpCodeVisitor;
         }
 
+        public CSharpCodeVisitor CSharpCodeVisitor { get; }
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpDesignTimeCodeVisitor.cs]: Why a property?

Comment (pranavkm) [inline src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpDesignTimeCodeVisitor.cs]: Mvc uses it

Comment (pranavkm) [inline src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpDesignTimeCodeVisitor.cs]: https://github.com/aspnet/Mvc/pull/3190/files#diff-1e2ae35d1a85838608fc4f2daf7c408bR49

Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpDesignTimeCodeVisitor.cs]: Talked in person, it don't :smile:

**Comment by NTaylorMullen** [comment]
Looks solid. :shipit: once MVC is signed off on.

**Pull request closed without merging**